### PR TITLE
CI Hwacha Support

### PR DESCRIPTION
**Pull request by abejgonzalez**
**Related issue**: Addressed part of #325 

**Type of change**: new feature

**Impact**: no rtl change

**Development Phase**: implementation

**Release Notes**
Added necessary changes to enable a Hwacha RoCC test in BOOM CI. Currently matches the CI test that Chipyard has. Also has small changes to enable not rebuilding the toolchain every time there is a Chipyard bump.

### CODE REVIEW

**Comment by abejgonzalez** [comment]
Holding off on this PR until https://github.com/ucb-bar/chipyard/pull/161 is resolved.

**Comment by abejgonzalez** [comment]
Includes small bugfix to PTW ports

File: CHIPYARD.hash [hunk 1]
```diff
@@ -1 +1 @@
-26042b682c8041b556ff5d83cd340242fbce12f6
\ No newline at end of file
+2f3ac85c2ae0a679cdd900df55dbf7f1998f638c
```
Comment (jerryz123) [inline CHIPYARD.hash:1]: What branch/commit of Chipyard is this?

File: .circleci/create-hash.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+cd $LOCAL_CHIPYARD_DIR
+
+echo "Checking out Chipyard version: $(cat $LOCAL_CHECKOUT_DIR/CHIPYARD.hash)"
+git fetch
+git checkout $(cat $LOCAL_CHECKOUT_DIR/CHIPYARD.hash)
+
+# Use normalized output of git-submodule status as hashfile
+for tools in 'riscv-tools' 'esp-tools' ; do
+    git submodule status "toolchains/${tools}" "toolchains/qemu" | while read -r line ; do
+        echo "${line#[!0-9a-f]}"
```
Comment (jerryz123) [inline .circleci/create-hash.sh:25]: This is to deal with the flattened riscv-tools in chipyard right?

**Review by jerryz123 — APPROVED** [review]

File: .circleci/create-hash.sh (hunk 2 above)
Comment (abejgonzalez) [reply .circleci/create-hash.sh:25]: Yes.

File: CHIPYARD.hash (hunk 1 above)
Comment (abejgonzalez) [reply CHIPYARD.hash:1]: It is 1 commit behind current Chipyard `dev`

**Pull request merged**